File: my-folder/0053-maximum-subarray/solution.java
class Solution {
    public int maxSubArray(int[] nums) {
        int prev = nums[0];
        int max = prev;
        for(int i = 1; i < nums.length; i++){
            if(prev > 0){
                prev += nums[i];
                max = Math.max(max, prev);
            } else {
                prev = nums[i];
                max = Math.max(max, prev);
            }
        }


        return max;
    }
}

File: my-folder/0053-maximum-subarray/solution.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        mSum = nums[0]
        s = 0
        i = 0

        for j in range(len(nums)):
            while i < j and s <= 0:
                s -= nums[i]
                i += 1

            s += nums[j]
            
            mSum = max(mSum, s)
        
        return mSum
